### PR TITLE
refactor: show search input only if workspace selector has more than 5 items

### PR DESCRIPTION
**Pull request by nastyastavitskaya**
## Summary
- Updating `WorkspaceSelector` to only show search input, if the selector has more than 5 items to enhance the UX.

## Testing Plan
- [x] Was this tested locally? If not, explain why.
- Run the story tests for GlobalNavigation with different numbers of items in the workspace selector.
- The new story test with interactions was added due to the search input disappearing bug when no results are found.

## Reference Issue (For mParticle employees only.  Ignore if you are an outside contributor)
- Closes https://mparticle-eng.atlassian.net/browse/UNI-220

### CODE REVIEW

**Review by jared-dickman — APPROVED** [review]
seems legit

**Comment by jared-dickman** [comment]
might need to adjust your branch name to have the checks pass

**Comment by nastyastavitskaya** [comment]
> might need to adjust your branch name to have the checks pass

so it's complaining about the `branch` name? I thought it was not happy with the PR title, which I changed since this check failed, but it is still unhappy 😞

**Review by tibuurcio — APPROVED** [review]

**Comment by jared-dickman** [comment]
would be great to add a test in a story 😉

File: src/components/navigation/GlobalNavigation/WorkspaceSelector/WorkspaceSelector.tsx [hunk 1]
```diff
@@ -112,6 +112,8 @@ export function WorkspaceSelector(props: IWorkspaceSelectorProps) {
 
   const workspaceInitials = getInitials(activeWorkspace?.label)
 
+  const hasSearchInput = menuItems.filter(item => !!item.label).length > 5
```
Comment (jared-dickman) [inline src/components/navigation/GlobalNavigation/WorkspaceSelector/WorkspaceSelector.tsx:115]: might want to add a check for an existing search term, the search bar seems to disappear when no results are found

**Review by jared-dickman — CHANGES_REQUESTED** [review]
**issue** search input disappears when no results are found

this will be considered a bug and should be addressed with a story test in this pr

File: src/components/navigation/GlobalNavigation/WorkspaceSelector/WorkspaceSelector.tsx (hunk 1 above)
Comment (nastyastavitskaya) [reply src/components/navigation/GlobalNavigation/WorkspaceSelector/WorkspaceSelector.tsx:115]: oof, yes! You're right, great 🐛 catch

**Comment by jared-dickman** [comment]
@tibuurcio this presents an interesting challenge, i cant check out this branch because its on a forked branch. 

@nastyastavitskaya it currently says in [Contributing](https://github.com/mParticle/aquarium/blob/c056d9f27b89096334f7570b64a123da2173d8fc/CONTRIBUTING.md#L1) to clone the repo, do you know where it said to fork? possibly that was changed recently?

**Comment by nastyastavitskaya** [comment]
> @tibuurcio this presents an interesting challenge, i cant check out this branch because its on a forked branch.
> 
> @nastyastavitskaya it currently says in [Contributing](https://github.com/mParticle/aquarium/blob/c056d9f27b89096334f7570b64a123da2173d8fc/CONTRIBUTING.md#L1) to clone the repo, do you know where it said to fork? possibly that was changed recently?

was changed 4 days ago:
Screenshot 2024-03-19 at 11 51 03 AM

**Comment by jared-dickman** [comment]
ah-ha makes sense

probably best for you to remove the forked repo, and clone instead, allowing the rest of the team to checkout the branches locally

**Comment by nastyastavitskaya** [comment]
> @tibuurcio this presents an interesting challenge, i cant check out this branch because its on a forked branch.
> 
> @nastyastavitskaya it currently says in [Contributing](https://github.com/mParticle/aquarium/blob/c056d9f27b89096334f7570b64a123da2173d8fc/CONTRIBUTING.md#L1) to clone the repo, do you know where it said to fork? possibly that was changed recently?

I guess I can close this PR, clone the repo instead of forking, and then create a new PR from it.

**Pull request closed without merging**